salt/returners/postgres_local_cache.py

```diff
@@ -232,7 +232,7 @@ def returner(load):
     sql = '''INSERT INTO salt_returns
             (fun, jid, return, id, success)
             VALUES (%s, %s, %s, %s, %s)'''
-    job_ret = {'return': six.text_type(six.text_type(load['return']), 'utf-8', 'replace')}
+    job_ret = {'return': six.text_type(load['return'])}
```

Comment:
This would break if the payload's return data contained encoded non-ascii unicode:

```python
>>> six.text_type('glücklich')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 2: ordinal not in range(128)
```

Also, if for some reason the function returned non-decodable bytes (such as this example here where I'm reading from `/dev/urandom`), then we'd get a similar error:

```python
>>> with open('/dev/urandom', 'rb') as f:
...     data = f.read(16)
...
>>> six.text_type(data)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xfa in position 0: ordinal not in range(128)
>>>
```

Therefore, how about something like this?

```python
try:
    ret = six.text_type(load['return'])
except UnicodeDecodeError:
    ret = str(load['return'])
job_ret = {'return': ret}
```

Reply:
So convert to unicode unless it will result in an error. In that case return the raw value. That makes sense to me.